test/auth.e2e-spec.ts

```diff
@@ -32,52 +38,55 @@ describe('AuthController (e2e)', () => {
     app = moduleRef.createNestApplication();
     prismaService = moduleRef.get<PrismaService>(PrismaService);
     authService = moduleRef.get<AuthService>(AuthService);
-    user = createUser();
-
+    twoFactorAuthRepository = moduleRef.get<TwoFactorAuthRepository>(
+      TwoFactorAuthRepository,
+    );
     app.useGlobalPipes(new ValidationPipe());
 
     await app.init();
   });
 
-  afterEach(async () => {
-    await prismaService.user.deleteMany({ where: { email: user.email } });
+  afterEach(() => {
+    prismaService.user.deleteMany();
   });
 
   describe('POST /auth/signup', () => {
+    const tempUser = createUser();
     it('should register a user and return the new user object', async () => {
       return request(app.getHttpServer())
         .post('/auth/signup')
         .set('Accept', 'application/json')
-        .send(user)
+        .send(tempUser)
         .expect((response: request.Response) => {
           const { id, email, role } = response.body;
           expect(id).toEqual(expect.any(Number));
-          expect(email).toEqual(user.email);
+          expect(email).toEqual(tempUser.email);
           expect(role).toEqual(Role.USER);
         })
         .expect(HttpStatus.CREATED);
     });
 
     it('should not register a user (already in db) and return 403 error (FORBIDDEN ACCESS)', async () => {
-      await prismaService.user.create({ data: user });
       return request(app.getHttpServer())
         .post('/auth/signup')
         .set('Accept', 'application/json')
-        .send(user)
+        .send(tempUser)
         .expect(HttpStatus.FORBIDDEN);
```

Comment:
hmm, kind of risky - imagine if someone puts another test in between those two that will remove `tempUser`. Also - not sure - but I think that if you run a second test without running the first one, it will fail. And thus it violates the principle that tests should be deterministic.

You should register a user in the first part of the test (just don't put too many expects as you are doing this in other test - just one to check if the user was indeed registered is enough)